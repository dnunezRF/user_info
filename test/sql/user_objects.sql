\set ECHO 0
BEGIN;
\i sql/user_info.sql
\set ECHO all

-- create empty role
CREATE ROLE foo;

-- nothing yet owned by foo
SELECT * FROM user_objects('foo');

-- create table and alter owner to foo
CREATE TABLE foo_table (id int);
ALTER TABLE foo_table OWNER TO foo;

-- nothing yet owned by foo - both queries should return 4 rows (types+relations)
SELECT count(*) FROM user_objects('foo');
SELECT count(*) FROM user_objects('foo') WHERE name = 'foo';

-- these two should return 1 (or 2 in case of TYPEs)
SELECT count(*) FROM user_objects('foo') WHERE type = 'RELATION';
SELECT count(*) FROM user_objects('foo') WHERE type = 'TABLE';
SELECT count(*) FROM user_objects('foo') WHERE type = 'TYPE';

ROLLBACK;
