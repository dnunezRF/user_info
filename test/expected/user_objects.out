\set ECHO 0
-- create empty role
CREATE ROLE foo;
-- nothing yet owned by foo
SELECT * FROM user_objects('foo');
 type | oid | schema | name 
------+-----+--------+------
(0 rows)

-- create table and alter owner to foo
CREATE TABLE foo_table (id int);
ALTER TABLE foo_table OWNER TO foo;
-- nothing yet owned by foo - both queries should return 4 rows (types+relations)
SELECT count(*) FROM user_objects('foo');
 count 
-------
     4
(1 row)

SELECT count(*) FROM user_objects('foo') WHERE name = 'foo';
 count 
-------
     0
(1 row)

-- these two should return 1 (or 2 in case of TYPEs)
SELECT count(*) FROM user_objects('foo') WHERE type = 'RELATION';
 count 
-------
     1
(1 row)

SELECT count(*) FROM user_objects('foo') WHERE type = 'TABLE';
 count 
-------
     1
(1 row)

SELECT count(*) FROM user_objects('foo') WHERE type = 'TYPE';
 count 
-------
     2
(1 row)

ROLLBACK;
